server/test/lib/device/device.getDeviceFeaturesStates.test.js

```diff
@@ -0,0 +1,91 @@
+const EventEmitter = require('events');
+const { expect, assert } = require('chai');
+const uuid = require('uuid');
+const { fake } = require('sinon');
+const db = require('../../../models');
+const Device = require('../../../lib/device');
+const Job = require('../../../lib/job');
+
+const event = new EventEmitter();
+const job = new Job(event);
+
+const now = new Date('2000-06-15T03:59:00.000Z');
+const insertStates = async () => {
+  const queryInterface = db.sequelize.getQueryInterface();
+  const deviceFeatureStateToInsert = [];
+  const statesToInsert = 3 * 60;
```

Comment:
I think this is still based from the tests from the aggregated states, but it doesn't seem to be really useful to have that much states ?

IMO, you're doing a simple function, you can write a simple test

In your case, you could even hardcode some states here (like 4 states) with fixed dates. 

Then, in your test, you write a query between fixed dates and make sure that only 2 of the 4 states are returned ? (for example)

If you want to test the ?take, you could add another test and do it with ?take=1 for example